File: Problemset/0720-Longest Word in Dictionary/Brute Force.py
class Solution:
    def longestWord(self, words: List[str]) -> str:
        words_set = set(words)
        ret = ""

        for word in words:
            temp = word
            while temp[:-1] in words_set:
                temp = temp[:-1]

            if not temp[:-1]:
                if len(word) > len(ret):
                    ret = word
                elif len(word) == len(ret) and word < ret:
                    ret = word

        return ret

File: Problemset/1408-String Matching in an Array/Brute Force.rs
impl Solution {
    pub fn string_matching(words: Vec<String>) -> Vec<String> {
        let mut ret = Vec::new();

        for i in 0..words.len() {
            for j in 0..words.len() {
                if j != i && words[j].contains(&words[i]) {
                    ret.push(words[i].clone());
                    break;
                }
            }
        }

        ret
    }
}
